runtime-sdk/src/context.rs

```diff
@@ -271,6 +275,31 @@ impl<'a, 'b> TxContext<'a, 'b> {
             .map(|x| x.downcast().expect("type should stay the same"))
             .unwrap_or_default()
     }
+
+    /// Fetches or sets a value associated with the transaction.
+    pub fn tx_value<V>(&mut self, key: &'static str) -> &mut V
+        where
+            V: Any + Default,
+    {
+        self.tx_values
+            .entry(key)
+            .or_insert_with(|| Box::new(V::default()))
```

Comment:
```suggestion
            .or_insert_with(Default::default)
```
should work

Reply:
Oh, in this case feel free to also fix the existing stuff.

Reply:
ok

Reply:
Oh you know what, it's inserting into `...Map<..., Box<dyn Any>>`, so all it knows is the `Any` in this context, so it doesn't know to use `V`

```
error[E0277]: the trait bound `(dyn std::any::Any + 'static): std::default::Default` is not satisfied
   --> runtime-sdk/src/context.rs:109:9
    |
109 | /         self.values
110 | |             .entry(key)
111 | |             .or_insert_with(Default::default)
    | |_____________________________________________^ the trait `std::default::Default` is not implemented for `(dyn std::any::Any + 'static)`
    | 
   ::: /home/wh0/.rustup/toolchains/nightly-2020-08-29-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/default.rs:84:1
    |
84  |   pub trait Default: Sized {
    |   ------------------------ required by this bound in `std::default::Default::default`
    |
    = note: required because of the requirements on the impl of `std::default::Default` for `std::boxed::Box<(dyn std::any::Any + 'static)>`

error[E0277]: the trait bound `dyn std::any::Any: std::default::Default` is not satisfied
   --> runtime-sdk/src/context.rs:111:29
    |
111 |             .or_insert_with(Default::default)
    |                             ^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `dyn std::any::Any`
    |
    = note: required because of the requirements on the impl of `std::default::Default` for `std::boxed::Box<dyn std::any::Any>`
    = note: required by `std::default::Default::default`
```